sdk/trace/span.go

```diff
@@ -36,6 +36,43 @@ const (
 	errorEventName  = "error"
 )
 
+// ReadOnlySpan allows reading information from the data structure underlying a
+// trace.Span. It is used in places where reading information from a span is
+// necessary but changing the span isn't necessary or allowed.
+// TODO: SpanContext(), Tracer() and IsRecording() are already required by
+// trace.Span. Does it make sense to require them here, too?
+// TODO: Should we make the methods unexported? The purpose of this interface
+// is controlling access to `span` fields, not having multiple implementations.
+type ReadOnlySpan interface {
+	Name() string
+	SpanContext() trace.SpanContext
+	Parent() trace.SpanContext
+	SpanKind() trace.SpanKind
+	StartTime() time.Time
+	EndTime() time.Time
+	Attributes() []label.KeyValue
+	Links() []trace.Link
+	// TODO: Does it make sense for Event to live under export?
```

Comment:
It does not make sense to me for it to live in the export package. I think it should be moved to the SDK.

Reply:
I'll resolve this since this is discussed in #1380 and in any case can't be done here due to the import cycle problem.